src/command_line_tool_lib.cc

```diff
@@ -0,0 +1,197 @@
+#include "command_line_tool_lib.h"
+
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <iostream>
+#include <stack>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "service_client_lib.h"
+
+// The definition of the service client
+ServiceClient command_tool::service_client;
+
+std::string command_tool::usage;
+
+ServiceClient::ReturnCodes command_tool::Register(const std::string &username) {
+  std::cout << "Registered username: " << username << ": ";
+
+  if (username.empty()) {
+    std::cout << "Empty username.\n";
+    std::cout << command_tool::usage;
+    return ServiceClient::INVALID_ARGUMENT;
+  }
+
+  // ServiceClient::ReturnCodes
+  auto ret = service_client.SendRegisterUserRequest(username);
+  std::cout << service_client.ErrorMsgs[ret] << "\n";
+
+  return ret;
+}
+
+ServiceClient::ReturnCodes command_tool::Chirp(const std::string &username,
+                                               const std::string &text,
+                                               const uint64_t &parent_id) {
+
+  std::cout << "Posted a chirp as " << username << ": ";
+
+  if (username.empty()) {
+    std::cout << "Empty username.\n";
+    std::cout << command_tool::usage;
+    return ServiceClient::INVALID_ARGUMENT;
+  }
+
+  if (text.empty()) {
+    std::cout << "Empty text.\n";
+    std::cout << command_tool::usage;
+    return ServiceClient::INVALID_ARGUMENT;
+  }
+
+  struct ServiceClient::Chirp chirp;
+  // ServiceClient::ReturnCodes
+  auto ret = service_client.SendChirpRequest(username, text, parent_id, &chirp);
+  std::cout << service_client.ErrorMsgs[ret] << "\n";
+  if (ret == ServiceClient::OK) {
+    std::cout << '\n';
+    command_tool::PrintSingleChirp(chirp, 0);
+  }
+
+  return ret;
+}
+
+ServiceClient::ReturnCodes command_tool::Follow(const std::string &username,
+                                                const std::string &to_follow) {
+
+  std::cout << "Followed " << to_follow << " as " << username << ": ";
+  if (username.empty()) {
+    std::cout << "Empty username.\n";
+    std::cout << command_tool::usage;
+    return ServiceClient::INVALID_ARGUMENT;
+  }
+
+  if (to_follow.empty()) {
+    std::cout << "Empty followee username.\n";
+    std::cout << command_tool::usage;
+    return ServiceClient::INVALID_ARGUMENT;
+  }
+
+  // ServiceClient::ReturnCodes
+  auto ret = service_client.SendFollowRequest(username, to_follow);
+  std::cout << service_client.ErrorMsgs[ret] << "\n";
```

Comment:
I've always seen std::endl instead of \n. Might be helpful for any cleanup

Reply:
I am not sure about this. What do you think? @barath